src/main/java/com/netflix/imflibrary/st2067_2/Application2E2021.java

```diff
@@ -0,0 +1,398 @@
+        }
+
+        return false;
+    }
+
+    public static void validateYUVImageCharacteristics(CompositionImageEssenceDescriptorModel imageDescriptor,
+            IMFErrorLogger logger) {
+        UUID imageDescriptorID = imageDescriptor.getImageEssencedescriptorID();
+
+        // Dimensions
+        if (imageDescriptor.getStoredWidth() > 3840 || imageDescriptor.getStoredWidth() < 1
+                || imageDescriptor.getStoredHeight() > 2160 || imageDescriptor.getStoredHeight() < 1) {
+            logger.addError(
+                    IMFErrorLogger.IMFErrors.ErrorCodes.APPLICATION_COMPOSITION_ERROR,
+                    IMFErrorLogger.IMFErrors.ErrorLevels.NON_FATAL,
+                    String.format(
+                            "EssenceDescriptor with ID %s has invalid image width or height per %",
+                            imageDescriptorID.toString(),
+                            APP_STRING));
+        }
+
+        // Sampling, colorimetry, color components and pixel depth
+        if (! isValidYUVColor(imageDescriptor)) {
+            logger.addError(
+                    IMFErrorLogger.IMFErrors.ErrorCodes.APPLICATION_COMPOSITION_ERROR,
+                    IMFErrorLogger.IMFErrors.ErrorLevels.NON_FATAL,
+                    String.format(
+                            "EssenceDescriptor with ID %s has invalid combination of sampling, colorimetry, color components and pixel depth as per %s",
+                            imageDescriptorID.toString(),
+                            APP_STRING));
+        }
+
+        // Quantization
+        if (!Quantization.QE1.equals(imageDescriptor.getQuantization())) {
+            logger.addError(
+                    IMFErrorLogger.IMFErrors.ErrorCodes.APPLICATION_COMPOSITION_ERROR,
+                    IMFErrorLogger.IMFErrors.ErrorLevels.NON_FATAL,
+                    String.format(
+                            "EssenceDescriptor with ID %s has invalid quantization as per %s",
+                            imageDescriptorID.toString(),
+                            APP_STRING));
+        }
+
+        // Sample rate
+        if (!FPS_UHD.contains(imageDescriptor.getSampleRate())) {
+            logger.addError(
+                    IMFErrorLogger.IMFErrors.ErrorCodes.APPLICATION_COMPOSITION_ERROR,
+                    IMFErrorLogger.IMFErrors.ErrorLevels.NON_FATAL,
+                    String.format("EssenceDescriptor with ID %s has invalid sample rate fas per %s",
+                            imageDescriptorID.toString(),
+                            APP_STRING));
+        }
+    }
+
+    public static boolean isValidRGBColor(CompositionImageEssenceDescriptorModel imageDescriptor) {
+        Integer depth = imageDescriptor.getPixelBitDepth();
+        Colorimetry color = imageDescriptor.getColor();
+        Quantization quantization = imageDescriptor.getQuantization();
+        Fraction fps = imageDescriptor.getSampleRate();
+
+        if (Colorimetry.Color4.equals(color)) {
```

Comment:
Color4 does not appear to be compatible with RGB color component encoding, please confirm.

Reply:
You are definitely right that COLOR.4 only supports YCbCr

Reply:
I am still wondering how this made it in there. It was a long time ago. Maybe I was looking at a draft image parameter table that had a bug?